ports/esp8266/network_wlan.c

```diff
@@ -170,6 +170,38 @@ static mp_obj_t esp_disconnect(mp_obj_t self_in) {
 }
 static MP_DEFINE_CONST_FUN_OBJ_1(esp_disconnect_obj, esp_disconnect);
 
+static mp_obj_t build_stations_tuple(void) {
+    mp_uint_t count = wifi_softap_get_station_num();
+    if (count == 0) {
+        return mp_const_empty_tuple;
+    }
+    mp_obj_tuple_t *stations = NULL;
+    struct station_info *info = wifi_softap_get_station_info();
+    if (info == NULL) {
+        mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("no station info available"));
+    }
+    nlr_buf_t nlr;
+    if (nlr_push(&nlr) == 0) {
```

Comment:
There's no need to wrap this in NLR.  Just let any exception propagate out.

Reply:
Ah, OK, so you need to clean up things if there's an exception.

But in that case you should still re-raise the exception, instead of printing it out.

You might be interested to look at `nlr_push_jump_callback()`, which is a relatively new feature to support this kind of use case.

Reply:
That indeed looks like it is what I needed there, thanks!  It also simplified the code by quite a bit too.

I've extended callback support to also accept functions with no arguments, but if the (tiny) size increase in the core runtime is not wanted I can always wrap the call to `wifi_softap_free_station_info` in a function that accepts (and discards) an argument.